cratedb_toolkit/wtf/library.py

```diff
@@ -0,0 +1,587 @@
+                    node['name'] AS node_name,
+                    COUNT(*) AS num_shards
+                FROM sys.shards
+                WHERE primary = true
+                GROUP BY node_name;
+            """,
+            description="Shard distribution across nodes.",
+        )
+
+        not_started = InfoElement(
+            name="shard_not_started",
+            label="Shards not started",
+            sql="""
+                SELECT *
+                FROM sys.allocations
+                WHERE current_state != 'STARTED';
+            """,
+            description="Information about shards which have not been started.",
+        )
+        not_started_count = InfoElement(
+            name="shard_not_started_count",
+            label="Number of shards not started",
+            description="Total number of shards which have not been started.",
+            sql="""
+                SELECT COUNT(*) AS not_started_count
+                FROM sys.allocations
+                WHERE current_state != 'STARTED';
+            """,
+            transform=get_single_value("not_started_count"),
+        )
+
+        rebalancing_progress = InfoElement(
+            name="shard_rebalancing_progress",
+            label="Shard Rebalancing Progress",
+            sql="""
+                SELECT
+                    table_name,
+                    schema_name,
+                    recovery['stage'] AS recovery_stage,
+                    AVG(recovery['size']['percent']) AS progress,
+                    COUNT(*) AS count
+                FROM
+                    sys.shards
+                GROUP BY table_name, schema_name, recovery_stage;
+            """,
+            description="Information about rebalancing progress.",
+        )
+
+        rebalancing_status = InfoElement(
+            name="shard_rebalancing_status",
+            label="Shard Rebalancing Status",
+            sql="""
+                SELECT node['name'], id, recovery['stage'], recovery['size']['percent'], routing_state, state
+                FROM sys.shards
+                WHERE routing_state IN ('INITIALIZING', 'RELOCATING')
+                ORDER BY id;
+            """,
+            description="Information about rebalancing activities.",
+        )
+
+        table_allocation = InfoElement(
+            name="table_allocation",
+            label="Table Allocations",
+            sql="""
+                SELECT
+                    table_schema, table_name, node_id, shard_id, partition_ident, current_state, decisions, explanation
+                FROM
+                    sys.allocations;
+            """,
+            description="Table allocation across nodes, shards, and partitions.",
+        )
+
+        table_allocation_special = InfoElement(
+            name="table_allocation_special",
+            label="Table Allocations Special",
+            sql="""
+                SELECT decisions[2]['node_name'] AS node_name, COUNT(*) AS table_count
```

Comment:
Why selecting the 2nd decision? This looks problematic e.g. when only 1 shard exists there isn't a 2nd decision.

Reply:
Also deferred to GH-169. Thanks.